### DISCUSSION TITLE
Styling fetched HTML correctly

### DISCUSSION THREAD

**Discussion by nerdess** [chakra-ui/chakra-ui#2404] [Q&A] [answered]
I am getting arbitrary content with HTML markup through Gatsby:

```
<h2>Hello magic unicorn</h2>
<p>Just some paragraph</p><p>Another paragraph</p>
```

When I drop this content on my site it is not styled with Chakra. Would I need to parse the HTML and replace e.g. the `<h2>`-Tags with `<Heading>`-Tags and the `<p>`-Tags with `<Text>`-Tags? 

What would be the correct approach here or is there a solution already build in Chakra on how to "chakrify" content streams?

Thx :)

**Comment by willem-aart** [comment, marked as answer]
Chakra's global styles (e.g. CSS reset) should be applied to all elements, so there's at least some styling going on, right?

You could indeed use something like [html-react-parser](https://github.com/remarkablemark/html-react-parser).

Another option is to render using a custom component (for example `HtmlContent`) with scoped  styling.

**Reply by ljosberinn** [reply]
Yup, that's the solution. Chakra doesn't style default html elements, it's a component library. It however resets CSS unless you opt out, so if anything, default html elements have less than normal styling. But for Chakra styles to apply, you must use `Heading` etc.

**Reply by nerdess** [reply]
@ljosberinn @willem-aart thx for confirming my gut feeling!

it just seems a bit crazy to replace e.g. a p-tag with `<Text>` just to get a p-tag again in the end but i understand why this is necessary :D

**Comment by VictorPeralta** [comment]
To anyone looking to solve this problem in the future, using Global Styles is also another possible solution

https://chakra-ui.com/docs/styled-system/global-styles
